<a href="https://colab.research.google.com/github/244107020170-png/machine-learning/blob/main/JS02_Assignment.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:
import pandas as pd
from sklearn.preprocessing import StandardScaler


In [2]:
from sklearn.datasets import load_breast_cancer

data = load_breast_cancer()
df = pd.DataFrame(data.data, columns=data.feature_names)
df['diagnosis'] = data.target  # 0 = malignant, 1 = benign


In [3]:
import pandas as pd
from sklearn.datasets import load_breast_cancer

# Load dataset dari sklearn
data = load_breast_cancer()

# Buat DataFrame
df = pd.DataFrame(data.data, columns=data.feature_names)

# Tambahkan kolom diagnosis (0 = malignant, 1 = benign)
df['diagnosis'] = data.target

# Cek 5 baris pertama
df.head()


,mean radius,mean texture,mean perimeter,mean area,mean smoothness,mean compactness,mean concavity,mean concave points,mean symmetry,mean fractal dimension,...,worst texture,worst perimeter,worst area,worst smoothness,worst compactness,worst concavity,worst concave points,worst symmetry,worst fractal dimension,diagnosis
0,17.99,10.38,122.80,1001.0,0.11840,0.27760,0.3001,0.14710,0.2419,0.07871,...,17.33,184.60,2019.0,0.1622,0.6656,0.7119,0.2654,0.4601,0.11890,0
1,20.57,17.77,132.90,1326.0,0.08474,0.07864,0.0869,0.07017,0.1812,0.05667,...,23.41,158.80,1956.0,0.1238,0.1866,0.2416,0.1860,0.2750,0.08902,0
2,19.69,21.25,130.00,1203.0,0.10960,0.15990,0.1974,0.12790,0.2069,0.05999,...,25.53,152.50,1709.0,0.1444,0.4245,0.4504,0.2430,0.3613,0.08758,0
3,11.42,20.38,77.58,386.1,0.14250,0.28390,0.2414,0.10520,0.2597,0.09744,...,26.50,98.87,567.7,0.2098,0.8663,0.6869,0.2575,0.6638,0.17300,0
4,20.29,14.34,135.10,1297.0,0.10030,0.13280,0.1980,0.10430,0.1809,0.05883,...,16.67,152.20,1575.0,0.1374,0.2050,0.4000,0.1625,0.2364,0.07678,0


In [4]:
print("Jumlah fitur:", len(data.feature_names))
print("Fitur usable:", data.feature_names)


Jumlah fitur: 30
Fitur usable: ['mean radius' 'mean texture' 'mean perimeter' 'mean area'
 'mean smoothness' 'mean compactness' 'mean concavity'
 'mean concave points' 'mean symmetry' 'mean fractal dimension'
 'radius error' 'texture error' 'perimeter error' 'area error'
 'smoothness error' 'compactness error' 'concavity error'
 'concave points error' 'symmetry error' 'fractal dimension error'
 'worst radius' 'worst texture' 'worst perimeter' 'worst area'
 'worst smoothness' 'worst compactness' 'worst concavity'
 'worst concave points' 'worst symmetry' 'worst fractal dimension']


In [5]:
# Ubah 0/1 jadi M/B kalau mau lihat
df['diagnosis_label'] = df['diagnosis'].map({0:'Malignant', 1:'Benign'})

# Encode diagnosis jadi 0/1 (M=0, B=1)
df['diagnosis'] = df['diagnosis_label'].map({'Malignant':0, 'Benign':1})

df[['diagnosis_label','diagnosis']].head()


,diagnosis_label,diagnosis
0,Malignant,0
1,Malignant,0
2,Malignant,0
3,Malignant,0
4,Malignant,0


In [6]:
from sklearn.preprocessing import StandardScaler

# Pisahkan fitur numerik
features = df.drop(columns=['diagnosis','diagnosis_label'])

# Standardisasi
scaler = StandardScaler()
scaled = scaler.fit_transform(features)

# Buat DataFrame hasil scaling
df_scaled = pd.DataFrame(scaled, columns=features.columns)

# Tambahkan diagnosis
df_scaled['diagnosis'] = df['diagnosis']

df_scaled.head()


,mean radius,mean texture,mean perimeter,mean area,mean smoothness,mean compactness,mean concavity,mean concave points,mean symmetry,mean fractal dimension,...,worst texture,worst perimeter,worst area,worst smoothness,worst compactness,worst concavity,worst concave points,worst symmetry,worst fractal dimension,diagnosis
0,1.097064,-2.073335,1.269934,0.984375,1.568466,3.283515,2.652874,2.532475,2.217515,2.255747,...,-1.359293,2.303601,2.001237,1.307686,2.616665,2.109526,2.296076,2.750622,1.937015,0
1,1.829821,-0.353632,1.685955,1.908708,-0.826962,-0.487072,-0.023846,0.548144,0.001392,-0.868652,...,-0.369203,1.535126,1.890489,-0.375612,-0.430444,-0.146749,1.087084,-0.243890,0.281190,0
2,1.579888,0.456187,1.566503,1.558884,0.942210,1.052926,1.363478,2.037231,0.939685,-0.398008,...,-0.023974,1.347475,1.456285,0.527407,1.082932,0.854974,1.955000,1.152255,0.201391,0
3,-0.768909,0.253732,-0.592687,-0.764464,3.283553,3.402909,1.915897,1.451707,2.867383,4.910919,...,0.133984,-0.249939,-0.550021,3.394275,3.893397,1.989588,2.175786,6.046041,4.935010,0
4,1.750297,-1.151816,1.776573,1.826229,0.280372,0.539340,1.371011,1.428493,-0.009560,-0.562450,...,-1.466770,1.338539,1.220724,0.220556,-0.313395,0.613179,0.729259,-0.868353,-0.397100,0


In [7]:
# Cek statistik setelah standardisasi
df_scaled.describe()

# Pastikan diagnosis tetap ada
df_scaled['diagnosis'].value_counts()


,count
diagnosis,
1,357
0,212


Report

**Introduction**

The Wisconsin Breast Cancer dataset consists of 569 records with diagnostic labels Malignant (M) and Benign (B). The purpose of preprocessing is to prepare the data for use in classification models.

**Dataset**

* Number of records: 569

* Number of features: 30 numerical (mean radius, mean texture, etc.)

* Label: diagnosis (M/B)


**Step 1 – Variable Identification**

* Usable: all numerical features.

* Non‑usable: no ID column in the sklearn version.

**Step 2 – Diagnosis Encoding**

* Diagnosis was converted into numeric values: Malignant = 0, Benign = 1.

**Step 3 – Standardization**

* All numerical features were standardized using StandardScaler.

* Result: mean ~0, standard deviation ~1.

**Step 4 – Validation**

* describe() shows that the feature distributions are standardized.

* value_counts() confirms that the diagnosis labels remain balanced (212 malignant, 357 benign).

**Conclusion**

The dataset is clean, the labels are properly encoded, and the numerical features are standardized. The dataset is now ready to be used for classification tasks in the next module.